src/arcaflow_plugin_sdk/test_schema.py

```diff
@@ -448,64 +448,77 @@ class TestSubclass(TestParent):
 
 
 class OneOfTest(unittest.TestCase):
-    def test_assignment(self):
-        @dataclasses.dataclass
-        class OneOfData1:
-            a: str
+    @dataclasses.dataclass
+    class OneOfData1:
+        a: str
 
-        @dataclasses.dataclass
-        class OneOfData2:
-            b: int
+    @dataclasses.dataclass
+    class OneOfData2:
+        b: int
 
-        scope = schema.ScopeType(
+    @dataclasses.dataclass
+    class OneOfDataEmbedded1:
+        type: str
+        a: str
+
+    def setUp(self):
+        self.obj_b: schema.ObjectType = schema.ObjectType(
+            OneOfTest.OneOfData2, {"b": PropertyType(schema.IntType())}
+        )
+        self.scope_basic: schema.ScopeType = schema.ScopeType(
             {
                 "a": schema.ObjectType(
-                    OneOfData1, {"a": PropertyType(schema.StringType())}
+                    OneOfTest.OneOfData1,
+                    {"a": PropertyType(schema.StringType())},
                 ),
-                "b": schema.ObjectType(
-                    OneOfData2, {"b": PropertyType(schema.IntType())}
+                "b": self.obj_b,
+            },
+            "a",
+        )
+        self.scope_embedded: schema.ScopeType = schema.ScopeType(
+            {
+                "a": schema.ObjectType(
+                    OneOfTest.OneOfDataEmbedded1,
+                    {
+                        "type": PropertyType(
+                            schema.StringType(),
+                        ),
+                        "a": PropertyType(schema.StringType()),
+                    },
                 ),
+                "b": self.obj_b,
             },
             "a",
         )
+
+    def test_assignment(self):
         s_type = schema.OneOfStringType(
-            {"a": schema.RefType("a", scope), "b": schema.RefType("b", scope)},
-            scope,
-            "_type",
+            {
+                "a": schema.RefType("a", self.scope_basic),
+                "b": schema.RefType("b", self.scope_basic),
+            },
+            scope=self.scope_basic,
+            discriminator_field_name="_type",
         )
         s_type.discriminator_field_name = "foo"
```

Comment:
Yeah, but I think that assignment was the purpose of the test, but I'm definitely not 100% sure. This test could be worth removing.

Reply:
Oh.  Weird.  I was assuming that the "assignment" being tested was in the invocation of `schema.OneOfIntType()`.  🥴

Otherwise, I think I agree that there's not much being tested here.